### PR TITLE
[requests] provide a service name for the request Span

### PR DESCRIPTION
**Pull request by palazzem**
### Overview

The current implementation doesn't provide any service name for the `requests` span. It means that all traces produced by `requests` as a root span will be dropped. With this patch we mainly:
* provide a default service name (that is `requests`), so that everytime it's a root span, the trace can be sent correctly
* allow users to change the service name for each client via:
```python
cfg = config.get_from(session)
cfg['service_name'] = 'client'
```
* the default service name can be changed also via environment variable `DD_REQUESTS_SERVICE_NAME` (or the legacy`DATADOG_REQUESTS_SERVICE_NAME`)
* if a parent is available and the service name has not been set by the user, use the parent service

During the process, minor refactoring has been done to split better the integration modules.

### Deprecated functionalities

The new configuration system has been used, but this patch is backward compatible so the legacy approach is still supported (but deprecated and will be removed in 1.0.0). Previously, to enable distributed tracing we had to:
```python
session.distributed_tracing = True
```
while now:
```python
cfg = config.get_from(session)
cfg['distributged_tracing'] = True
```

### CODE REVIEW

**Comment by palazzem** [comment]
Before proceeding it's better to implement a generic way to store integration settings because it's awkward to add Datadog attributes to the connection itself. It's way better to have:
* a generic configuration object
* a specific configuration object that doesn't pollute the wrapped object; in that case the API must be easy to use, without requiring any interaction with the `Pin` object.

File: ddtrace/contrib/requests/legacy.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+from ddtrace import config
```
[the remark this answers is not in the document]
Comment (LotharSee) [reply ddtrace/contrib/requests/legacy.py]: Should we add "deprecation" logging messages?

[the remark this answers is not in the document]
Comment (palazzem) [reply ddtrace/contrib/requests/legacy.py]: good point! adding that... the sooner the better.

File: ddtrace/contrib/requests/__init__.py [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
 
     from ddtrace.contrib.requests import TracedSession
 
     session = TracedSession()
-    session.get("http://www.datadog.com")
+    session.get("https://www.datadoghq.com")
 
-To enable distributed tracing, for example if you call, from requests, a web service
-which is also instrumented and want to have traces including both client and server sides::
+The library can be configured globally and per instance, using the Configuration API::
```
Comment (palazzem) [inline ddtrace/contrib/requests/__init__.py]: The end goal, is having a dedicated section to explain how you set options. Then this part will list only available options.

**Pull request merged**